packages/gasket-redux/src/configure-make-store.js

```diff
@@ -34,25 +34,25 @@ export function prepareReducer(allReducers, rootReducer) {
  * @type {import('./index').configureMakeStore}
  */
 export default function configureMakeStore(makeStoreOptions = {}, postCreate) {
-  const {
-    thunkMiddleware = thunk,
-    middleware = [],
-    logging = false,
-    enhancers = [(f) => f],
-    reducers = {},
-    rootReducer,
-    initialState = {}
-  } = makeStoreOptions;
-
-  const baseMiddleware = [thunkMiddleware];
-
   /**
    * Wrapper for store create to create instance with SSR and to hydrate in
    * browser.
    * @type {import('./index').MakeStoreFn}
    */
+  // eslint-disable-next-line max-statements
   function makeStore(state = {}, options = {}) {
     const { req, logger = new Log() } = options;
+    const {
+      thunkMiddleware = thunk,
+      middleware = [],
+      logging = false,
+      enhancers = [(f) => f],
+      reducers = {},
+      rootReducer,
+      initialState = {}
+    } = typeof makeStoreOptions === 'function' ? makeStoreOptions(req) : makeStoreOptions;
```

Comment:
Would it be valuable to pass all the `makeStore` options through?
```suggestion
    } = typeof makeStoreOptions === 'function' ? makeStoreOptions(options) : makeStoreOptions;
```